core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/SimulatedAnealingMinimizer.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.pipeline.movavg;
+
+import com.google.common.collect.EvictingQueue;
+import org.elasticsearch.search.aggregations.pipeline.movavg.models.MovAvgModel;
+
+/**
+ * A cost minimizer which will fit a MovAvgModel to the data.
+ *
+ * This optimizer uses naive simulated annealing.  Random solutions in the problem space
+ * are generated, compared against the last period of data, and the least absolute deviation
+ * is recorded as a cost.
+ *
+ * If the new cost is better than the old cost, the new coefficients are chosen.  If the new
+ * solution is worse, there is a temperature-dependent probability it will be randomly selected
+ * anyway.  This allows the algo to sample the problem space widely.  As iterations progress,
+ * the temperature decreases and the algorithm rejects poor solutions more regularly,
+ * theoretically honing in on a global minimum.
+ */
+public class SimulatedAnealingMinimizer {
+
+    /**
+     * Runs the simulated annealing algorithm and produces a new set of model coefficients.
+     * The method is quasi-generic, accepting any number of coefficients and trusting that
```

Comment:
I think this comment is a bit out of date as the method now accepts the model itself rather than a set of coefficients.